shared/src/commonMain/kotlin/io/customer/shared/di/KMMPlatformComponent.kt

```diff
@@ -0,0 +1,24 @@
+package io.customer.shared.di
+
+import io.customer.shared.Platform
+import io.customer.shared.database.DatabaseDriverFactory
+import io.customer.shared.database.getDatabaseDriverFactory
+
+/**
+ * Platform component dependency graph to satisfy platform based dependencies from single place.
+ *
+ * The class should only contain dependencies matching the following criteria:
+ * - dependencies that requires platform classes to initialize (e.g. Android Context).
+ * - dependencies that are should be defined/initialized once in SDK lifetime irrespective of
+ * workspace changes.
+ *
+ * @see [KMMStaticComponent] for complete graph hierarchy.
+ */
+@Suppress("MemberVisibilityCanBePrivate", "unused")
+internal class KMMPlatformComponent(
+    private val staticComponent: KMMStaticComponent,
+    val platform: Platform,
+) : DIGraph() {
```

Comment:
See other PR comment about multiple DiGraphs.

Reply:
This was already planned to be removed as we didn't see its benefit for KMM for now, I just missed it. I'll be doing that soon.